packages/cli/src/lib/upload/upload-command.unit.test.ts

```diff
@@ -4,20 +4,32 @@ import { beforeEach, describe, expect, it } from 'vitest';
 import { PortalUploadArgs, uploadToPortal } from '@code-pushup/portal-client';
 import {
   ISO_STRING_REGEXP,
+  MEMFS_VOLUME,
   MINIMAL_REPORT_MOCK,
 } from '@code-pushup/testing-utils';
 import { DEFAULT_CLI_CONFIGURATION } from '../../../mocks/constants';
 import { yargsCli } from '../yargs-cli';
 import { yargsUploadCommandObject } from './upload-command';
 
+// mock bundleRequire inside importEsmModule used for fetching config
+vi.mock('bundle-require', async () => {
+  const { CORE_CONFIG_MOCK }: typeof import('@code-pushup/testing-utils') =
+    await vi.importActual('@code-pushup/testing-utils');
+  return {
+    bundleRequire: vi
+      .fn()
+      .mockResolvedValue({ mod: { default: CORE_CONFIG_MOCK } }),
+  };
+});
```

Comment:
What if we extracted this to a helper function in `testing-utils`, with `CORE_CONFIG_MOCK` as an overridable default? :thinking:

Something like this:

- default config content:
  ```ts
  import { mockBundleRequire } from `@code-pushup/testing-utils`;
  
  mockBundleRequire();
  
  describe('...', () => {
    // ...
  })
  ```;
- custom config content:

  ```ts
  import { mockBundleRequire, CORE_CONFIG_MOCK } from `@code-pushup/testing-utils`;
  
  mockBundleRequire(({ filepath }) => ({
    ...CORE_CONFIG_MOCK,
    persist: {
      outputDir: `tmp/${getExtension(filepath)}`,
    },
  }));
  
  describe('...', () => {
    // ...
  });
  ```

:top: Just a suggestion, not sure how well this can work with `vi.mock` because of hoisting. And I could also understand the counter-argument that maybe adding an abstraction layer is less transparent (depends a lot on how robust the abstraction is, I think).

Reply:
The primary reason for splitting it was to directly add the config value to the same file as the tests (while previously one had to go to `testing-utils`).

So if it was a helper function where you would pass it the value you want it to return, sure. But here you need to deal with inserting value based on `filepath` into `mod: { default: SMTH } }`... I wouldn't complicate it for now.